src/libraries/Microsoft.PowerFx.Core/Public/Values/InMemoryTableValue.cs

```diff
@@ -16,7 +17,7 @@ internal class InMemoryTableValue : CollectionTableValue<DValue<RecordValue>>
         private readonly RecordType _recordType;
 
         internal InMemoryTableValue(IRContext irContext, IEnumerable<DValue<RecordValue>> records)
-            : base(irContext, records)
+            : base(irContext, records.ToList())
```

Comment:
This is ok to enable mutable semantics immediately, but we should find some way to unblock.